app/main.py

```diff
@@ -1 +1,42 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    @classmethod
+    def is_alive(cls, animal: annotations) -> None:
+        if animal in cls.alive:
+            if animal.health <= 0:
+                cls.alive.remove(animal)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}," \
+               f" Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden:
+            self.hidden = False
+        else:
+            self.hidden = True
```

Comment:
<img width="826" alt="image" src="https://github.com/mate-academy/py-herbivores-and-carnivores/assets/93856423/ad220858-cac1-4985-95b8-ac283e4bdb04">